mllib/src/main/scala/org/apache/spark/ml/tree/impl/GradientBoostedTrees.scala

```diff
@@ -311,6 +311,7 @@ private[spark] object GradientBoostedTrees extends Logging {
     val validationTol = boostingStrategy.validationTol
     treeStrategy.algo = OldAlgo.Regression
     treeStrategy.impurity = OldVariance
+    require(!treeStrategy.bootstrap, "GradientBoostedTrees does not need bootstrap sampling")
```

Comment:
can we do this checking inside next line ```treeStrategy.assertValid()```?

Reply:
I am afraid not, since this method is called in DecisionTree/GBT/RF, and bootstrap should be always False for DecisionTree and GBT while can be True in RF.